File: golden/javascript/find_maximum_product.js
function solution(X, B) {
    const N = B.length;
    if (N === 0) {
        return 0;
    }
    if (N < X) {
        return 0;
    }
    if (N === 1) {
        return B[0];
    }

    var max = 0;

    for (var i = X; i <= N; i++) {
        var range = B.slice(i - X, i);
        var prod = range.reduce(((a, e) => a * e), 1);
        if (prod > max) {
            max = prod;
        }
    }

    return max;
}

File: golden/python/find_maximum_product.py
from functools import reduce

def solution(X, B):
    N = len(B)
    if N == 0:
        return 0
    if N < X:
        return 0
    if N == 1:
        return B[0]

    max = 0
    for i in range(X, N + 1):
        window = B[i-X:i]
        prod = reduce((lambda a, b: a * b), window, 1)
        if max < prod:
            max = prod

    return max
